controller/api/destination/watcher/ip_watcher.go

```diff
@@ -168,12 +242,13 @@ func (ss *serviceSubscriptions) updateService(service *corev1.Service) {
 		Name:      service.Name,
 	}
 
-	if id != ss.id {
+	if id != ss.service {
 		for listener, port := range ss.listeners {
-			ss.endpoints.Unsubscribe(ss.id, port, "", listener)
+			ss.endpoints.Unsubscribe(ss.service, port, "", listener)
 			ss.endpoints.Subscribe(id, port, "", listener)
```

Comment:
Maybe this questions is quite silly, but just trying to understand things a little better. Why is this error not handled here. Is it because the `Svc().Informer()` is guaranteed to not return services of type `ExternalName`? What happens if this service changes to an external type right before you hit the k8s API?

Reply:
This is not a silly question, it's actually a great point.

We should probably handle errors here by sending a `NoEndpoints(exists = false)` message to the subscribers.